### PR TITLE
Optionally redirect metadata server traffic.

### PR DESCRIPTION
**Pull request by Q-Lee**
**What this PR does / why we need it**: hide node component keys from user pods.

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by Q-Lee** [comment]
cc @mikedanese
cc @thockin

**Comment by Q-Lee** [comment]
cc @cjcullen

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -48,6 +48,9 @@ function config-ip-firewall {
     iptables -A FORWARD -w -p UDP -j ACCEPT
     iptables -A FORWARD -w -p ICMP -j ACCEPT
   fi
+  if [[ -n "${BLACKHOLE_MDS}" ]]; then
```
Comment (mikedanese) [inline cluster/gce/gci/configure-helper.sh]: This is going to break. You need a ${BLACKHOLE_MDS:-}.

File: cluster/gce/gci/configure-helper.sh [hunk 2]
```diff
@@ -48,6 +48,9 @@ function config-ip-firewall {
...
+    iptables -t nat -I PREROUTING -p tcp -d 169.254.169.254 --dport 80 -j DNAT --to-destination 127.0.0.1:2020
```
Comment (thockin) [inline cluster/gce/gci/configure-helper.sh]: What is at 127.0.0.1:2020 ?  This should at least explain what is happening?  Given the name, I would expect this to jump to REJECT or DROP

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (thockin) [inline cluster/gce/gci/configure-helper.sh]: Also, is "MDS" a common reduction of "metadata server" ?  It's not in my vernacular...

**Comment by gmarek** [comment]
As you have @thockin and @mikedanese on this I won't be reviewing this:)

**Comment by ethernetdan** [comment]
Removing PR from 1.6 milestone since code freeze has begun:
* If this is a bug fix, please label as `kind/bug` and reapply the milestone.
* If this is a feature for 1.6, an [exception](https://github.com/kubernetes/features/blob/master/EXCEPTIONS.md) should be filed for it before 11:00PM PST on 3/2. Note that most exceptions to code freeze are not accepted.
* If this is a feature for 1.7, feel free to add the 1.7 milestone to the PR.

/cc @kubernetes/release-team @kubernetes/kubernetes-release-managers

**Comment by calebamiles** [comment]
This PR also does not have justification for why it should be considered a release blocker. Removing this label. Please do not add a `release-blocker` label without consulting with the [release team](https://github.com/kubernetes/features/blob/master/release-1.6/release_team.md). Thanks!

**cc: @ethernetdan, @kubernetes/kubernetes-release-managers**

**Comment by Q-Lee** [comment]
A colleague got this work in while I was sick.

**Pull request closed without merging**